File: frontend/turnosys-web/src/api/empresa.ts
import apiClient from './client'
import type { ApiResponse } from '@/types'

export interface Empresa {
  id: string
  razonSocial: string
  nombreFantasia?: string
  cuit?: string
  direccion?: string
  telefono?: string
  email?: string
  logotipoUrl?: string
  sitioWeb?: string
  instagram?: string
  facebook?: string
  whatsApp?: string
  horarioDesde?: string
  horarioHasta?: string
  timeZone: string
  observaciones?: string
  isActivo: boolean
  tieneLogo: boolean
}

export type EmpresaForm = Omit<Empresa, 'id' | 'isActivo' | 'tieneLogo' | 'logotipoUrl'>

export const empresaApi = {
  get: async () => {
    const { data } = await apiClient.get<ApiResponse<Empresa>>('/empresa')
    return data.data
  },
  actualizar: async (form: EmpresaForm) => {
    await apiClient.put('/empresa', form)
  },
  subirLogo: async (archivo: File) => {
    const fd = new FormData()
    fd.append('archivo', archivo)
    await apiClient.post('/empresa/logo', fd, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
  },
}

File: src/TurnoSys.Domain/Entities/Empresa.cs
using TurnoSys.Domain.Entities.Common;

namespace TurnoSys.Domain.Entities;

public class Empresa : BaseEntity
{
    public string RazonSocial { get; set; } = string.Empty;
    public string? NombreFantasia { get; set; }
    public string? CUIT { get; set; }
    public string? Direccion { get; set; }
    public string? Telefono { get; set; }
    public string? Email { get; set; }
    public string? LogotipoUrl { get; set; }
    public byte[]? LogotipoBlob { get; set; }
    public string? LogotipoContentType { get; set; }
    public string? SitioWeb { get; set; }
    public string? Instagram { get; set; }
    public string? Facebook { get; set; }
    public string? WhatsApp { get; set; }
    public TimeOnly? HorarioDesde { get; set; }
    public TimeOnly? HorarioHasta { get; set; }
    public string TimeZone { get; set; } = "America/Argentina/Buenos_Aires";
    public string? Observaciones { get; set; }
    public bool IsActivo { get; set; } = true;

    public ICollection<Profesional> Profesionales { get; set; } = [];
    public ICollection<Paciente> Pacientes { get; set; } = [];
    public ICollection<Practica> Practicas { get; set; } = [];
    public ICollection<Usuario> Usuarios { get; set; } = [];
    public ICollection<ParametroSistema> Parametros { get; set; } = [];
}
